packages/commerce-sdk-react/src/hooks/useUsid.ts

```diff
@@ -5,27 +5,30 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 import useAuthContext from './useAuthContext'
-import useLocalStorage from './useLocalStorage'
-import useConfig from './useConfig'
 
-const onClient = typeof window !== 'undefined'
+/**
+ * @group Shopper Authentication helpers
+ */
+interface Usid {
+    usid: string | null
+    getUsidWhenReady: () => Promise<string>
+}
 
 /**
  * Hook that returns the usid associated with the current access token.
  *
  * @group Helpers
  * @category Shopper Authentication
  */
-const useUsid = (): string | null => {
-    const config = useConfig()
+const useUsid = (): Usid => {
     const auth = useAuthContext()
+    const usid = auth.get('usid')
+
+    // NOTE: auth.ready() is to be called later. If you call it immediately in this hook,
+    // it'll cause infinite re-renders during testing.
```

Comment:
The code is pretty self-explanatory.

```suggestion
```

Reply:
Vincent was mimicking the comment found in other similar files. IIRC I was the original author of that comment.

It was not obvious to me that it could cause infinite rerendering (and I think it happened during testing only). And that was why I wrote the comment.

Reply:
Yes, I based that code on what's in `useAccessToken` and had copied over those comments from there.

Reply:
😛 IMO This is pretty basic react hooks concept, it's nothing specific to our implementation, that's why i don't think we need a comment.

react hooks are called on every re-render, and you don't want to call anything that could potentially have side effects, you must do that in an event callback or `useEffect`. This is the same as you wouldn't want to call an async function in react component's render method.

Reply:
i guess the name of the function `auth.ready` might be confusing, it does have side effect!